### PR TITLE
[PowerPC] Mark llvm.tan on vectors as expand

### PR DESCRIPTION
**Pull request by ecnelises**

### CODE REVIEW

**Comment by farzonl** [comment]
Hi thanks for picking this up iI've been working on a general fix so I have a list of all the tests that need to be updated. For PowerPC it is:

- CodeGen/PowerPC/f128-arith.ll
- CodeGen/PowerPC/lower-intrinsics-afn-mass.ll
- CodeGen/PowerPC/lower-intrinsics-afn-mass_notail.ll
- CodeGen/PowerPC/lower-intrinsics-fast-mass.ll
- CodeGen/PowerPC/lower-intrinsics-fast-mass_notail.ll
- CodeGen/PowerPC/lower-intrinsics-nofast-mass.ll
- CodeGen/PowerPC/unal-altivec2.ll
- CodeGen/PowerPC/lower-intrinsics-fast-mass.ll
- CodeGen/PowerPC/lower-intrinsics-fast-mass_notail.ll
- CodeGen/PowerPC/lower-intrinsics-nofast-mass.ll
- CodeGen/PowerPC/unal-altivec2.ll

Also in `llvm/include/llvm/Analysis/VecFuncs.def  add
```cpp
TLI_DEFINE_VECFUNC("llvm.tan.f64", "__tand2", FIXED(2), "_ZGV_LLVM_N2v") 
TLI_DEFINE_VECFUNC("llvm.tan.f32", "__tanf4", FIXED(4), "_ZGV_LLVM_N4v")
```

File: llvm/lib/Target/PowerPC/PPCISelLowering.cpp [hunk 1]
```diff
@@ -860,6 +860,7 @@ PPCTargetLowering::PPCTargetLowering(const PPCTargetMachine &TM,
       setOperationAction(ISD::FEXP2, VT, Expand);
       setOperationAction(ISD::FSIN, VT, Expand);
       setOperationAction(ISD::FCOS, VT, Expand);
+      setOperationAction(ISD::FTAN, VT, Expand);
```
Comment (chenzheng1030) [inline llvm/lib/Target/PowerPC/PPCISelLowering.cpp:863]: Do we need to handle STRICT_FTAN too?

Comment (ecnelises) [reply llvm/lib/Target/PowerPC/PPCISelLowering.cpp:863]: It's already been handled by common logic, and there's already been test for constrained tan in vector-constrained-fp-intrinsics.ll

**Comment by ecnelises** [comment]
Hi @farzonl, do you mean this patch is not needed, because you will fix vector version of these math library nodes together?

**Comment by farzonl** [comment]
> Hi @farzonl, do you mean this patch is not needed, because you will fix vector version of these math library nodes together?

My plan was to do an expand in `TargetLoweringBase.cpp` in this pr: https://github.com/llvm/llvm-project/pull/95517 That way I knock out more platforms at once.

I'm doing spot testing to  make sure f16 and vector work. Also PowerPC needs llvm vectors to convert to IBM MASS  Vector functions and  my change doesn't do that either.

**Pull request closed without merging**